src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -58,21 +52,50 @@ public static Prefix[] getPrefixesPresent(ArgumentMultimap argMultimap) {
+     * Create Patient from prefixes present
      *
-     * @param argMultimap Argument multimap which contains prefix to value mapping
-     * @param prefixes    List of prefixes present
-     * @return Patient object
+     * @param argMultimap Contains mapping of key which is prefix and value which is argument value
+     * @param prefixes    List of prefixes present in argument
+     * @return Patient with the fields present in user input
      * @throws ParseException if the user input does not conform the expected format
      */
     public static Patient createPatientFromPrefixes(ArgumentMultimap argMultimap, Prefix[] prefixes)
-            throws ParseException {
+        throws ParseException {
+        // filling the fields with default values
         Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
-        Phone phone = new Phone("12345678");
-        Email email = new Email("default_email@gmail.com");
-        Address address = new Address("Address not added");
+        Phone phone = new Phone(Phone.DEFAULT_PHONE);
+        Email email = new Email(Email.DEFAULT_EMAIL);
+        Gender gender = new Gender(Gender.DEFAULT_GENDER);
+        IcNumber icNumber = new IcNumber(IcNumber.DEFAULT_IC_NUMBER);
+        Birthday birthday = new Birthday(Birthday.DEFAULT_BIRTHDAY);
+        Address address = new Address(Address.DEFAULT_ADDRESS);
         Set<Tag> tagList = new HashSet<>();
+
+        // passing to helper function to replace fields with actual values if it exists
+        return createPatientFromPresentPrefixes(name, phone, email, gender, icNumber, birthday, address, tagList,
+            argMultimap, prefixes);
+    }
+
+    /**
+     * Replaces Patient fields with actual value if it is present in argMultimap
+     *
+     * @param name        Name of patient
+     * @param phone       Phone number of patient
+     * @param email       Email of patient
+     * @param gender      Gender of Patient
+     * @param icNumber    IcNumber of Patient
+     * @param birthday    Birthday of Patient
+     * @param address     Address of Patient
+     * @param tags        Tags of Patient
+     * @param argMultimap Contains mapping of key which is prefix and value which is argument value
+     * @param prefixes    List of prefixes present in argument
+     * @return Patient with the fields present in user input
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public static Patient createPatientFromPresentPrefixes(Name name, Phone phone, Email email, Gender gender,
+                                                           IcNumber icNumber, Birthday birthday, Address address,
+                                                           Set<Tag> tags, ArgumentMultimap argMultimap,
+                                                           Prefix[] prefixes) throws ParseException {
         for (Prefix p : prefixes) {
```

Comment:
Also I'm not sure whether I am tripping or missing something but I don't see the cases for the other prefixes here? 🤣 Or will they only be added later?

Reply:
Yup they will be added later!! Would require the ParserUtil class to be completed before i can continue the implementation.

Reply:
Yup everything looks good! Definitely agree with the changes Jingya suggested! Points of focus for the next iteration: 

1. Including the remaining prefixes to be parsed to create patient objects
2. Finding a way to load and save tasks for storage of patient info.
3. Updating the Emergency Department where - is used for inactive patients rather than new patients!

Reply:
Got it thanks!